### DISCUSSION TITLE
Can't accept messages for closing gripper due to differ md5sum in message type

### DISCUSSION THREAD

**Discussion by ItoMasaki** [HumanSupportRobot/QandA#41] [Q&A] [answered]
I am currently working on the code to close the gripper and am trying to verify its operation through ihsrb (gripper.command(-0.2) or gripper.apply_force(10)). The problem I have encountered is that the latest ros-noetic-tmc-desktop-full does not allow action communication because the md5sum of the message does not match on the server side and the client side. However, the old ros-noetic-tmc-desktop-full was able to close the gripper properly.　We would appreciate it if you could provide us with a solution.

Here are the details of the environment.
HSR
>RELEASE_VERSION="21.12"
ROBOT_VERSION=HSRB-PHASE5
LOCALE=ja_JP.UTF-8
BUILD_DATE=20210319_000000_+0000

Private computer
>ros-noetic-desktop-full : 1.5.0-1focal.20221209.231810
ros-noetic-tmc-desktop-full : 0.18.0-1.focal.20221026.0040.+0000

The following is a description of the warnings.

> [WARN] [1677082233.157647]: Could not process inbound connection: Client [/hsrb/robot_hardware] wants topic [/hsrb/gripper_controller/grasp/goal] to have datatype/md5sum [tmc_control_msgs/GripperApplyEffortActionGoal/43355ef33a9f9ba5d26e50a3cc6e60bd], but our version has [tmc_control_msgs/GripperApplyEffortActionGoal/09d77e47aded4b89a3aa6a2d855822a3] Dropping connection.{'callerid': '/hsrb/robot_hardware', 'md5sum': '43355ef33a9f9ba5d26e50a3cc6e60bd', 'tcp_nodelay': '0', 'topic': '/hsrb/gripper_controller/grasp/goal', 'type': 'tmc_control_msgs/GripperApplyEffortActionGoal'}
[WARN] [1677082237.186229]: Could not process inbound connection: Client [/hsrb/robot_hardware] wants topic [/hsrb/gripper_controller/apply_force/goal] to have datatype/md5sum [tmc_control_msgs/GripperApplyEffortActionGoal/43355ef33a9f9ba5d26e50a3cc6e60bd], but our version has [tmc_control_msgs/GripperApplyEffortActionGoal/09d77e47aded4b89a3aa6a2d855822a3] Dropping connection.{'callerid': '/hsrb/robot_hardware', 'md5sum': '43355ef33a9f9ba5d26e50a3cc6e60bd', 'tcp_nodelay': '0', 'topic': '/hsrb/gripper_controller/apply_force/goal', 'type': 'tmc_control_msgs/GripperApplyEffortActionGoal'}

**Comment by ItoMasaki** [comment, marked as answer]
I solved myself this issue.
In your client computer

1. Open _GripperApplyEffortActionGoal.py in /opt/ros/noetic/lib/python3/
dist-packages/tmc_control_msgs/msg
2. Replace _md5sum from "09d77e47aded4b89a3aa6a2d855822a3" to "43355ef33a9f9ba5d26e50a3cc6e60bd"
3. Check to there are no warnings by opening ihsrb

**Comment by MatthijsBurgh** [comment]
You should make sure both systems have the same version of `tmc_control_msgs`. The solution you applied just overrules the safety checks ROS put in place to prevent issues arising from mismatches in msg definitions.

So `roscd tmc_control_msgs` on both systems. Check `package.xml` for the version and make sure these are the same. Otherwise you need to upgrade the software on one of these systems.

**Reply by ayuguchi** [reply]
Thanks for your mention. I also agree to avoid the above overrules the safety checks on ROS.
Here, I also have faced the same problem. I just followed your suggestion to check the `package.xml` between my client's PC and an HSR, but the version is the same. For the next step, it seems that I should upgrade the software on the HSR. Here, I would like to know if officially approved upgrading the software on an HSR by myself or not.
